src/App.tsx

```diff
@@ -2,28 +1,54 @@
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+  const items = getNumbers(1, 42).map(n => `Item ${n}`);
+  const total = items.length;
+  const itemsPerPageOptions = [3, 5, 10, 20];
+
+  const handlePageChange = (page: number) => {
+    setCurrentPage(page);
+  };
+
+  const handlePerPageChange = (newPerPage: number) => {
+    setPerPage(newPerPage);
+    setCurrentPage(1);
+  };
+
+  const startIndex = Math.min((currentPage - 1) * perPage + 1, total);
+  const endIndex = Math.min(startIndex + perPage - 1, total);
```

Comment:
The calculation for 'startIndex' and 'endIndex' is done on every render. This could be optimized by using useMemo to only recalculate when 'currentPage' or 'perPage' changes.